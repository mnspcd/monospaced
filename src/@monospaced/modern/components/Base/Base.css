@import url("../../assets/fonts/index.css");

@media (--smaller-viewport) {
  body.mdrn-is-overlayed {
    overflow: hidden;
  }
}

.mdrn {
  background-color: var(--color-white);
  color: var(--color-grey-9);
  font-family: var(--font-body-text);
  font-size: 1rem;
  font-variant: common-ligatures contextual;
  line-height: var(--leading);
  text-rendering: optimizeLegibility;
}

@media (--larger-viewport) {
  .mdrn {
    font-size: 1.25rem;
  }
}

@media (--largest-viewport) {
  .mdrn {
    font-size: 1.5rem;
  }
}

.mdrn *,
.mdrn ::after,
.mdrn ::before {
  box-sizing: border-box;
}

.mdrn a {
  color: var(--color-blue-6);
  text-decoration: none;
}

.mdrn a:hover {
  color: var(--color-blue-7);
  outline: 0;
}

.mdrn a:active {
  color: var(--color-blue-8);
  outline: 0;
}

.mdrn b,
.mdrn h1,
.mdrn h2,
.mdrn h3,
.mdrn h4,
.mdrn h5,
.mdrn h6,
.mdrn strong,
.mdrn th {
  font-weight: 400; /* never bold */
}

.mdrn code {
  font-family: var(--font-monospaced);
  font-size: calc(1em * var(--type-scale-s));
  font-variant: slashed-zero;
}

.mdrn pre {
  font-family: var(--font-monospaced);
  margin: 0;
}

.mdrn small {
  font-size: calc(1em * var(--type-scale-m));
}

@media (--larger-viewport) {
  .mdrn small {
    font-size: calc(1em * var(--type-scale-s));
  }
}

.mdrn sub {
  font-variant-position: sub;
}

.mdrn sup {
  font-variant-position: super;
}

.mdrn time {
  font-family: var(--font-monospaced);
  font-size: calc(1em * var(--type-scale-m));
  font-variant: slashed-zero;
}

@media (--larger-viewport) {
  .mdrn time {
    font-size: calc(1em * var(--type-scale-s));
  }
}

.mdrn .u-monospaced {
  font-family: var(--font-monospaced);
  font-variant: slashed-zero;
}

.mdrn .u-smallCaps {
  font-family: var(--font-small-caps);
  text-transform: lowercase;
}
