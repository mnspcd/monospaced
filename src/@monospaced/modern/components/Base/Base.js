import PropTypes from "prop-types";
import React from "react";

import "./Base.css";

const Base = ({ children }) => <div className="mdrn">{children}</div>;

Base.propTypes = {
  children: PropTypes.any,
};

export default Base;
