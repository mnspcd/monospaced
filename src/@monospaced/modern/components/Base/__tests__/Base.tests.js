/* eslint-env jest */
import { render } from "@testing-library/react";
import React from "react";

import Base from "..";

describe("Base component", () => {
  it("should render correctly", () => {
    const { container } = render(<Base>children</Base>);

    expect(container).toMatchSnapshot();
  });
});
