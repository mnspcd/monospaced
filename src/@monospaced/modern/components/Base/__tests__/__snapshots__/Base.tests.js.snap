// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base component should render correctly 1`] = `
<div
  className="mdrn"
>
  children
</div>
`;
